### ISSUE TITLE
Assert error in VFP_Open(), cache/cache_fetch_proc.c line 139: Condition((vc->wrk->vsl) != 0) not true

### ISSUE THREAD

**Issue by nigoroll** [varnishcache/varnish-cache#2104]
This is master as of May, at first sight this appears unfixed to me - but have only spent 10 Minutes yet.

interesting bits:
- The URL starts with `/http`

```
Panic at: Thu, 29 Sep 2016 02:48:15 GMT
"Assert error in VFP_Open(), cache/cache_fetch_proc.c line 139:
  Condition((vc->wrk->vsl) != 0) not true.
thread = (cache-worker)
version = varnish-trunk revision bff2e6d
ident = Linux,2.6.32-642.1.1.el6.x86_64,x86_64,-junix,-smalloc,-smalloc,-hcritbit,epoll
Backtrace:
  0x432c73: pan_ic+0x163
  0x424a3e: VFP_Open+0xfe
  0x434d07: vrb_pull+0xa7
  0x43565c: VRB_Ignore+0x3c
  0x436f07: cnt_lookup+0x477
  0x437531: CNT_Request+0x3e1
  0x44f89f: HTTP1_Session+0x55f
  0x44ff07: http1_req+0x57
  0x449ccd: WRK_Thread+0x49d
  0x44a0fb: pool_thread+0x2b
req = 0x7f1b076a8020 {
  vxid = 431003709, transport = HTTP/1 {
    state = HTTP1::Proc
  }
  step = R_STP_LOOKUP,
  req_body = R_BODY_TAKEN,
  restarts = 0, esi_level = 0,
  sp = 0x7f1b92ce2020 {
    fd = 20, vxid = 431003708,
    t_open = 1475117294.106567,
    t_idle = 1475117294.106567,
    transport = HTTP/1 {
      state = HTTP1::Proc
    }
client = *REDACTED* 24294,
  },
  worker = 0x7f1e3a4c2bd0 {
    stack = {0x7f1e3a4c3000 -> 0x7f1e3a4b7000},
    ws = 0x7f1e3a4c2db8 {
      id = \"wrk\",
      {s, f, r, e} = {0x7f1e3a4c2390, 0x7f1e3a4c2390, (nil), +2040},
    },
    VCL::method = HIT,
    VCL::return = deliver,
    VCL::methods = {HIT},
  },
  ws = 0x7f1b076a81f8 {
    id = \"req\",
    {s, f, r, e} = {0x7f1b076a9ff8, +1184, (nil), +57344},
  },
  http_conn = 0x7f1b076a8128 {
    fd = 20,
    doclose = NULL,
    ws = 0x7f1b076a81f8,
    {rxbuf_b, rxbuf_e} = {0x7f1b076a9ff8, 0x7f1b076aa2b6},
    {pipeline_b, pipeline_e} = {0x7f1b076aa2b6, 0x7f1b076aa316},
    content_length = 96,
    body_status = length,
    first_byte_timeout = 0.000000,
    between_bytes_timeout = 0.000000,
  },
  http[req] = 0x7f1b076a8290 {
    ws[req] = 0x7f1b076a81f8,
    hdrs {
      \"POST\",
      \"/http://*REDACTED*/\",
      \"HTTP/1.1\",
      \"Host: *REDACTED*\",
      \"Content-Length: 96\",
      \"Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8\",
      \"Cf-Connecting-Ip: *REDACTED*\",
      \"Cf-Ipcountry: FR\",
      \"Cf-Origin-Https: off\",
      \"Cf-Ray: *REDACTED*\",
      \"Cf-Visitor: {\"scheme\":\"http\"}\",
      \"Content-Type: application/x-www-form-urlencoded\",
      \"Referer: http://*REDACTED*/\",
      \"X-Forwarded-Proto: http\",
      \"Connection: close\",
      \"X-Forwarded-For: *REDACTED*, *REDACTED*\",
      \"X-Real-IP: *REDACTED*\",
      \"Accept-Encoding: gzip\",
    },
  },
  vcl = {
    busy = 11
    discard = 0,
    state = auto,
    temp = warm,
    conf = {
      srcname = {
        \"/etc/varnish/default.vcl\",
        \"Builtin\",
      },
    },
  },
  objcore[BODY] = 0x7f1a758394c0 {
    refcnt = 1,
    flags = 0x22,
    exp_flags = 0x0,
    boc = 0x7f1a67b968c0 {
      refcnt = 1,
      state = 0,
      vary = (nil),
      stevedore_priv = (nil),
    },
    exp = {0.000000, 0.000000, 0.000000, 0.000000}
    objhead = 0x7f1e39cf12b0,
    stevedore = 0x7f1e39ce8180 (malloc Transient) {
      Simple = 0x7f1a5524b380
      Obj 0x7f1b093fcdd8 {priv=0x7f1b093fcdd0, ptr=0x7f1a5524b380, len=120, space=128}
      LEN = 0x0...0
      VXID = 0x00000000
      FLAGS = 0x00
      GZIPBITS = 0x0...0
      LASTMODIFIED = 0x0...0
      VARY = {len=0, ptr=(nil)}
      HEADERS = {len=0, ptr=(nil)}
    },
  },
  objcore[REQ] = 0x7f1a997a77a0 {
    refcnt = 2,
    flags = 0x0,
    exp_flags = 0x4,
    exp = {1475117294.422020, 1800.000000, 7200.000000, 0.000000}
    objhead = 0x7f1a95bf49b0,
    stevedore = 0x7f1e39ce80c0 (malloc s0) {
      Simple = 0x7f1ac9098800
      Obj 0x7f1ab02f7d88 {priv=0x7f1ab02f7d80, ptr=0x7f1ac9098800, len=485, space=488}
      LEN = 0x0...0
      VXID = 0x1a1615b0
      FLAGS = 0x00
      GZIPBITS = 0x0...0
      LASTMODIFIED = 0x41d5fb203b800000
      VARY = {len=0, ptr=0x7f1ac9098878}
      HEADERS = {len=0, ptr=0x7f1ac909888d}
    },
  },
  flags = {
  },
},

"

```

**Comment by nigoroll** [comment]
I started at the code for quite some time and fail to see any reason why `vc->wrk->vsl` (which I assume to be `req->wrk->vsl`) should be NULL for anything called from `CNT_Request()`.

**Comment by nigoroll** [comment]
has not occurred again, I'll blame vmware for the time being.

**Comment by nigoroll** [comment]
Occurred again, different commit, different project, no vmware

```
Panic at: Fri, 11 Nov 2016 11:50:35 GMT
"Assert error in VFP_Open(), cache/cache_fetch_proc.c line 138:
  Condition((vc->wrk->vsl) != 0) not true.
thread = (cache-worker)
version = varnish-5.0.0 revision 4eee5c4
ident = Linux,2.6.32-504.8.1.el6.x86_64,x86_64,-junix,-smalloc,-smalloc,-hcritbit,epoll
Backtrace:
  0x434313: pan_ic+0x163
  0x425b9e: VFP_Open+0xfe
  0x4363b7: vrb_pull+0xa7
  0x436d2c: VRB_Ignore+0x3c
  0x438570: cnt_lookup+0x3f0
  0x438bf1: CNT_Request+0x411
  0x4545cf: HTTP1_Session+0x4ef
  0x454da7: http1_req+0x57
  0x44c4e5: WRK_Thread+0x495
  0x44c91b: pool_thread+0x2b
req = 0x7fac943a8020 {
  vxid = 460636289, transport = HTTP/1 {
    state = HTTP1::Proc
  }
  step = R_STP_LOOKUP,
  req_body = R_BODY_TAKEN,
  restarts = 0, esi_level = 0,
  sp = 0x7fac299c6020 {
    fd = 83, vxid = 460636288,
    t_open = 1478865033.533247,
    t_idle = 1478865033.533247,
    transport = HTTP/1 {
      state = HTTP1::Proc
    }
client = 127.0.0.1 55748,
  },
  worker = 0x7fb3b494ebd0 {
    stack = {0x7fb3b494f000 -> 0x7fb3b4943000},
    ws = 0x7fb3b494edb8 {
      id = \"wrk\",
      {s, f, r, e} = {0x7fb3b494e390, 0x7fb3b494e390, (nil), +2040},
    },
    VCL::method = HIT,
    VCL::return = deliver,
    VCL::methods = {HIT},
  },
  ws = 0x7fac943a81f8 {
    id = \"req\",
    {s, f, r, e} = {0x7fac943aa658, +2896, (nil), +55712},
  },
  http_conn = 0x7fac943a8128 {
    fd = 83,
    doclose = NULL,
    ws = 0x7fac943a81f8,
    {rxbuf_b, rxbuf_e} = {0x7fac943aaa58, 0x7fac943aad0f},
    {pipeline_b, pipeline_e} = {0x7fac943aad0f, 0x7fac943aae3b},
    content_length = 898,
    body_status = length,
    first_byte_timeout = 0.000000,
    between_bytes_timeout = 0.000000,
  },
  http[req] = 0x7fac943a8290 {
    ws[req] = 0x7fac943a81f8,
    hdrs {
      \"POST\",
      \"/*REDACTED*\",
      \"HTTP/1.1\",
      \"POST\",
      \"...REDACTED...\",
      \"X-Varnish2-Varnish4-Proxy: true\",
      \"X-Varnish: 1799717813\",
      \"...REDACTED...\"
    },
  },
  vcl = {
    name = \"vcl_20161109125251\"
    busy = 54
    discard = 0,
    state = auto,
    temp = warm,
    conf = {
      srcname = {
        \"...REDACTED...\",
      },
    },
  },
  objcore[BODY] = 0x7fae080c46a0 {
    refcnt = 1,
    flags = {busy, private},
    exp_flags = {},
    boc = 0x7fac34205880 {
      refcnt = 1,
      state = invalid,
      vary = (nil),
      stevedore_priv = (nil),
    },
    exp = {0.000000, 0.000000, 0.000000, 0.000000}
    objhead = 0x7fb3da8cb390,
    stevedore = 0x7fb3da8f1180 (malloc Transient) {
      Simple = 0x7fb260489c00
      Obj 0x7fac833681f8 {priv=0x7fac833681f0, ptr=0x7fb260489c00, len=120, space=128}
      LEN = 0x0...0
      VXID = 0x00000000
      FLAGS = 0x00
      GZIPBITS = 0x0...0
      LASTMODIFIED = 0x0...0
      VARY = {len=0, ptr=(nil)}
      HEADERS = {len=0, ptr=(nil)}
    },
  },
  objcore[REQ] = 0x7fac365f7fc0 {
    refcnt = 2,
    flags = {},
    exp_flags = {refd},
    exp = {1478865033.759263, 60.000000, 10.000000, 0.000000}
    objhead = 0x7fac463821d0,
    stevedore = 0x7fb3da8f10c0 (malloc s0) {
      Simple = 0x7fac629fff00
      Obj 0x7fac1775bc98 {priv=0x7fac1775bc90, ptr=0x7fac629fff00, len=1580, space=1584}
      LEN = 0x0000000000006c3a
      VXID = 0x1b44d81b
      FLAGS = 0x00
      GZIPBITS = 0x0...0
      LASTMODIFIED = 0x41d6096c22400000
      VARY = {len=0, ptr=0x7fac629fff78}
      HEADERS = {len=0, ptr=0x7fac629fffac}
      Body 0x7fac1775bf68 {priv=0x7fac1775bf60, ptr=0x7faeae774000, len=16384, space=16384}
      Body 0x7fac1776b158 {priv=0x7fac1776b150, ptr=0x7fae667c5000, len=11322, space=11322}
    },
  },
  flags = {
  },
},

"
```

**Comment by nigoroll** [comment]
more panics with the same message and stack trace seen with 4eee5c4 on
* Wed, 07 Dec 2016 15:03:20 GMT
* Wed, 21 Dec 2016 22:49:46 GMT
* Sun, 25 Dec 2016 18:22:31 GMT

**Comment by hermunn** [comment]
Backport review: This is master only, as noted in the fix #2210.
